pkg/server/controllers.go

```diff
@@ -680,6 +632,70 @@ func (s *Server) installAPIBindingController(ctx context.Context, config *rest.C
 	})
 }
 
+func (s *Server) installAPIBinderController(ctx context.Context, config *rest.Config, server *genericapiserver.GenericAPIServer) error {
+	// Client used to create APIBindings within the initializing workspace
+	config = rest.CopyConfig(config)
+	kcpclienthelper.SetMultiClusterRoundTripper(config)
+	config = rest.AddUserAgent(config, initialization.ControllerName)
+	config.Host = fmt.Sprintf("https://%v/services/%v/%v", s.GenericConfig.ExternalAddress, initializingworkspaces.VirtualWorkspaceName, tenancyv1alpha1.ClusterWorkspaceAPIBindingsInitializer)
+	initializingWorkspacesKcpClusterClient, err := kcpclient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	// Wildcard client used for informers
+	informerCfg := rest.CopyConfig(config)
+	kcpclienthelper.SetCluster(informerCfg, logicalcluster.Wildcard)
+	informerClient, err := kcpclient.NewForConfig(informerCfg)
+	if err != nil {
+		return err
+	}
+	initializingWorkspacesKcpInformers := kcpexternalversions.NewSharedInformerFactoryWithOptions(
+		informerClient,
+		resyncPeriod,
+		kcpexternalversions.WithExtraClusterScopedIndexers(indexers.ClusterScoped()),
+		kcpexternalversions.WithExtraNamespaceScopedIndexers(indexers.NamespaceScoped()),
+	)
+
+	c, err := initialization.NewAPIBinder(
+		initializingWorkspacesKcpClusterClient,
+		initializingWorkspacesKcpInformers.Tenancy().V1alpha1().ClusterWorkspaces(),
+		s.KcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaceTypes(),
+		s.KcpSharedInformerFactory.Apis().V1alpha1().APIBindings(),
+		s.KcpSharedInformerFactory.Apis().V1alpha1().APIExports(),
+	)
+	if err != nil {
+		return err
+	}
+
+	return server.AddPostStartHook(postStartHookName(initialization.ControllerName), func(hookContext genericapiserver.PostStartHookContext) error {
+		logger := klog.FromContext(ctx).WithValues("postStartHook", postStartHookName(initialization.ControllerName))
+
+		if err := s.waitForSync(hookContext.StopCh); err != nil {
+			logger.Error(err, "failed to finish post-start-hook")
+			return nil // don't klog.Fatal. This only happens when context is cancelled.
+		}
+
+		initializingWorkspacesKcpInformers.Start(hookContext.StopCh)
+		initializingWorkspacesKcpInformers.WaitForCacheSync(hookContext.StopCh)
+
+		if err := wait.PollImmediateInfiniteWithContext(goContext(hookContext), time.Millisecond*100, func(ctx context.Context) (bool, error) {
+			synced := initializingWorkspacesKcpInformers.Tenancy().V1alpha1().ClusterWorkspaces().Informer().HasSynced() &&
+				s.KcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaceTypes().Informer().HasSynced() &&
+				s.KcpSharedInformerFactory.Apis().V1alpha1().APIBindings().Informer().HasSynced() &&
+				s.KcpSharedInformerFactory.Apis().V1alpha1().APIExports().Informer().HasSynced()
```

Comment:
why this loop? for this we have `waitForSync` above.

Reply:
will remove